### PR TITLE
Describe and hoist stripped state to a first-class citizen

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/matrix-org/matrix-doc/issues/3413
MSC: https://github.com/matrix-org/matrix-doc/pull/3173

Preview: https://pr3606--matrix-org-previews.netlify.app

### CODE REVIEW

File: content/client-server-api/_index.md [hunk 1]
```diff
@@ -1399,6 +1399,65 @@ following field:
...
+* [`m.room.encryption`](#mroomencryption)
+
+{{% boxes/rationale %}}
+These state events represent basic aesthetic information about the room
+and give clients information about how to join the room, and what to
+expect once joined.
+
+The name, avatar, topic, and aliases are presented as aesthetic information
+about the room, allowing users to make decisions about whether or not they
+want to join the room.
```
Comment (clokep) [inline content/client-server-api/_index.md:1438]: These two paragraphs sound duplicative right next to each other. (I do like that it gives an overview and then more detailed info, but the entire section seems short enough that maybe you can just jump into it?)

File: content/client-server-api/_index.md [hunk 2]
```diff
@@ -1399,6 +1399,65 @@ following field:
...
+Clients should only use stripped state events so long as they don't have
+access to the proper state of the room. Once the state of the room is
+available, all stripped state should be discarded. In cases where the
+client has an archived state of the room (such as after being kicked)
+and the client is receiving stripped state for the room, such as from an
+invite or knock, then the stripped state should take precedence until
+fresh state can be acquired from a join.
+
+The following state events should be represented as stripped state when
+possible:
```
Comment (clokep) [inline content/client-server-api/_index.md:1421]: Maybe it is clear from this, but should it be made explicit that not all events might be sent by the server (or available in the room)? I.e. clients shouldn't assume all of the below will exist?

**Review by uhoreg — APPROVED** [review]
Assuming @clokep doesn't have further suggestions, lgtm

**Review by clokep — APPROVED** [review]
No more suggestions from me, very happy to see this formally declared in a single spot!

**Pull request merged**